fastavro/_write_common.py

```diff
@@ -0,0 +1,18 @@
+
+class ValidationException(Exception):
+    def __init__(self, datum, schema, field):
+        message = '{field} is {datum} of type ' \
```

Comment:
We might want to omit the `{field} is` if `field` is `None`. When messing around I got an error like `None is None of type <NoneType> expected str` which was initially a little confusing.

Reply:
Ya I was going for simple approach to start, so ya maybe when updating ValidationError object, it can customize the message for None.

Reply:
Single error.
```
  File "/app/fastavro/fastavro/_validate_py.py", line 175, in validate
    raise ValidationErrors(ValidationError(datum, schema, ns_field))
fastavro._validate_common.ValidationErrors: [
  "namespace.missingerror.str is 11 of type <class 'int'> expected string"
]
```

Multiple errors on same union field.
```
  File "/app/fastavro/fastavro/_validate_py.py", line 126, in validate_union
    raise ValidationErrors(*errors)
fastavro._validate_common.ValidationErrors: [
  "namespace.missingerror.str_null is 11 of type <class 'int'> expected null",
  "namespace.missingerror.str_null is 11 of type <class 'int'> expected string"
]
```

Reply:
Should ValidationErrors be changed to ValidationException?

Reply:
Ugh, I'm kind of torn... I would say to keep `ValidationErrors` as it's more similar to normal types of exceptions (e.g. `TypeError`, `ValueError`, `KeyError`, `OSError`, etc), but I wouldn't want someone to make a typo by omitting the final `s`.

I guess I'm slightly still in favor of `ValidationErrors` but could be convinced otherwise.